repositories.bzl

```diff
@@ -174,3 +174,266 @@ def transcoding_repositories(bind=True):
         )
+
+def zlib_repositories(bind=True):
+    BUILD = """
+# Copyright 2016 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+#
+
+licenses(["notice"])
+
+exports_files(["README"])
+
+cc_library(
+    name = "zlib",
+    srcs = [
+        "adler32.c",
+        "crc32.c",
+        "crc32.h",
+        "deflate.c",
+        "deflate.h",
+        "infback.c",
+        "inffast.c",
+        "inffast.h",
+        "inffixed.h",
+        "inflate.c",
+        "inflate.h",
+        "inftrees.c",
+        "inftrees.h",
+        "trees.c",
+        "trees.h",
+        "zconf.h",
+        "zutil.c",
+        "zutil.h",
+    ],
+    hdrs = [
+        "zlib.h",
+    ],
+    copts = [
+        "-Wno-shift-negative-value",
+        "-Wno-unknown-warning-option",
+    ],
+    defines = [
+        "Z_SOLO",
+    ],
+    visibility = [
+        "//visibility:public",
+    ],
+)
+"""
+
+    native.new_git_repository(
+        name = "zlib_git",
+        build_file_content = BUILD,
+        commit = "50893291621658f355bc5b4d450a8d06a563053d",  # v1.2.8
+        remote = "https://github.com/madler/zlib.git",
+    )
+
+    native.bind(
+        name = "zlib",
+        actual = "@zlib_git//:zlib"
+    )
+
+
+def grpc_repositories(bind=True, envoy_deps=False):
+    zlib_repositories(bind)
+
+    native.git_repository(
+        name = "grpc_git",
+        commit = "c80d3321d0f77bef8cfff8b32490a07c1e90a5ad", # v1.3.2
+        remote = "https://github.com/grpc/grpc.git",
+    )
+
+    native.bind(
+        name = "nanopb",
+        actual = "@grpc_git//third_party/nanopb",
+    )
+
+    if envoy_deps:
+        native.bind(
+            name = "cares",
+            actual = "@envoy_deps//:ares",
+        )
+
+    if bind:
+        native.bind(
+            name = "gpr",
+            actual = "@grpc_git//:gpr",
+        )
+
+        native.bind(
+            name = "grpc",
+            actual = "@grpc_git//:grpc",
+        )
+
+        native.bind(
+            name = "grpc_cpp_plugin",
+            actual = "@grpc_git//:grpc_cpp_plugin",
+        )
+
+        native.bind(
+            name = "grpc++",
+            actual = "@grpc_git//:grpc++",
+        )
+
+        native.bind(
+            name = "grpc_lib",
+            actual = "@grpc_git//:grpc++_codegen_proto",
+        )
+
+def googleapis_repositories(protobuf_repo="@protobuf_git//", bind=True):
```

Comment:
https://github.com/istio/proxy/blob/master/src/envoy/transcoding/BUILD#L54

transcoding is using the repo, not function.

To avoid confusion, moved the loading part to top level WORKSPACE.